### ISSUE TITLE
feature: Tag a variable as containing a glob

### ISSUE THREAD

**Issue by LukeShu** [koalaman/shellcheck#1013]
#### For new checks and feature suggestions
- [x] shellcheck.net (i.e. the latest commit) currently gives no useful warnings about this
- [x] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

In a program I am working on, one of the configuration variables is a glob for acceptable file extensions.  Currently, shellcheck gives me warnings about leaving it unquoted--which I need to do.  On the other
hand, if I *do* quote it, *that* is probably a mistake; except in a few special cases, like with `find`.  I would like a way to be able to inform shellcheck that the variable is expected to contain a glob, and that it should generally flip the its opinion on whether it should be quoted.

#### Here's a snippet or screenshot that shows the problem:

config.sh:
```sh
#!/hint/bash

PKGDIR='/path/to/pkg dir'

# shellcheck vartype=glob (or something)
PKGEXT='.pkg.tar.?z'
```

script:
```sh
#!/bin/bash
. config
# wrong: PKGDIR needs quoted
files=($PKGDIR/*$PKGEXT)
printf ' -> %s\n' "${files[@]}"

# wrong: PKGEXT needs unquoted
files=("$PKGDIR/"*"$PKGEXT")
printf ' -> %s\n' "${files[@]}"

# just right
files=("$PKGDIR/"*$PKGEXT)
printf ' -> %s\n' "${files[@]}"

# wrong: PKGEXT glob needs quoted for find
find "$PKGDIR" -name '*'$PKGEXT

# just right
find "$PKGDIR" -name "*$PKGEXT"
```

#### Here's what shellcheck currently says:

```
Line 4:
files=($PKGDIR/*$PKGEXT)
       ^-- SC2206: Quote to prevent word splitting, or split robustly with mapfile or read -a.
                ^-- SC2206: Quote to prevent word splitting, or split robustly with mapfile or read -a.
 
Line 12:
files=("$PKGDIR/"*$PKGEXT)
                  ^-- SC2206: Quote to prevent word splitting, or split robustly with mapfile or read -a.

Line 16:
find "$PKGDIR" -name '*'$PKGEXT
                        ^-- SC2086: Double quote to prevent globbing and word splitting.
```

#### Here's what I wanted to see:

```
Line 4:
files=($PKGDIR/*$PKGEXT)
       ^-- SC2206: Quote to prevent word splitting, or split robustly with mapfile or read -a.

Line 7:
files=("$PKGDIR/"*"$PKGEXT")
                  ^-- SC2206: Unquote to enable globbing

Line 16:
find "$PKGDIR" -name '*'$PKGEXT
                        ^-- SC2086: Double quote to prevent globbing and word splitting.
```
